### PR TITLE
[golz4]add lz4 streaming API

### PR DESCRIPTION
**Pull request by zzzzssss**
1. Add all of the lz4 C files in `src`
2. Write the Go interface for lz4 streaming API from this example:
https://github.com/lz4/lz4/blob/master/examples/blockStreaming_doubleBuffer.c

3. Add unit test and bench mark for Lz4 streaming API

### CODE REVIEW

File: src/dll/example/Makefile [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# ##########################################################################
```
Comment (alexandreolivier-dd) [inline src/dll/example/Makefile]: The whole `dll` folder should probably not be there

Comment (zzzzssss) [reply src/dll/example/Makefile]: 👍

File: lz4_test.go [hunk 2]
```diff
@@ -192,3 +210,168 @@ func TestFuzz(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestContinueCompress(t *testing.T) {
+	payload := []byte("Hello World!")
+	repeat := 2
```
Comment (zzzzssss) [inline lz4_test.go]: when `repeat` is more than 1, this test will fail. The `LZ4_decompress_safe_continue` always returns an error code -16. Googled the problem, and tried to make the size of the destination buffer larger, still has error.  

Found this related issue, but the usage is different from our case.
https://github.com/Metaswitch/sas-client/issues/68

File: lz4_test.go [hunk 3]
```diff
@@ -192,3 +210,168 @@ func TestFuzz(t *testing.T) {
...
+	n, err := r.Read(dst)
+	if err != io.EOF {
+		t.Fatalf("Error should have been EOF, was %s instead: (%v bytes read: %s)", err, n, dst[:n])
+	}
+	failOnError(t, "Failed to close decompress object", r.Close())
+
+}
+
+func TestStreamSimpleCompressionDecompression(t *testing.T) {
+	inputs, _ := ioutil.ReadFile("sample2.txt")
```
Comment (zzzzssss) [inline lz4_test.go:248]: when the `len(inputs)` is longer than `RecommendBlockSize` (which is 8kb now), it will fail.

**Pull request merged**